src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -928,7 +928,14 @@ std::string onDetectRmdSourceType(
                                        "<!-- rmarkdown v1 -->") &&
           rmarkdownPackageAvailable())
       {
-         return "rmarkdown";
+         if (boost::algorithm::icontains(pDoc->contents(), "html_notebook"))
```

Comment:
I think we should be a bit more careful in this check, since IIUC this would trigger if `html_notebook` were mentioned anywhere in the document (even outside the YAML header).

1. Could we parse and check the YAML directly for the output type? (Can we assume the document's YAML header is in a "good" / parseable state here?) This would also protect us from the myriad of ways users might decide to encode the YAML string here (quoted vs unquoted and so on).

2. Could we use a regular expression to more specifically check for `runtime: html_notebook` in the document (handling the arbitrary whitespace separating the tokens)? That would at least be safer than just checking for this string only. We might also want to be robust against usages like `output: rmarkdown::html_notebook` or `output: "html_notebook"`.

Reply:
That's a good point. I don't think we want to be as aggressive as actually parsing YAML here since this runs so frequently. I've updated this so it only looks in the YAML header. (This is congruent with how we e.g. detect `runtime: shiny` documents.) The odds of having `html_notebook` in your YAML header if it's not a notebook are vanishingly small.